storage/cache/subtree_cache.go

```diff
@@ -321,44 +308,39 @@ func (s *SubtreeCache) getNodeHash(id tree.NodeID, getSubtree GetSubtreeFunc) ([
 		if err != nil {
 			glog.Errorf("base64.DecodeString(%v): %v", sfxKey, err)
 		}
-		glog.Infof("getNodeHash(%x | %x): %x", subKey, b, nh)
+		glog.Infof("getNodeHash(%x | %x): %x", subID, b, nh)
 	}
 
 	if glog.V(3) {
-		glog.Infof("cache: getNodeHash(path=%x, prefixLen=%d) => %x}", id.Path, id.PrefixLenBits, nh)
+		glog.Infof("cache: getNodeHash(%+v) => %x}", id, nh)
 	}
 	return nh, nil
 }
 
 // SetNodeHash sets a node hash in the cache.
-func (s *SubtreeCache) SetNodeHash(nID compact.NodeID, h []byte, getSubtree GetSubtreeFunc) error {
-	id, err := tree.NewNodeIDForTreeCoords(int64(nID.Level), int64(nID.Index), maxSupportedTreeDepth)
-	if err != nil {
-		return fmt.Errorf("failed to create nodeID %v: %v", nID, err)
-	}
+func (s *SubtreeCache) SetNodeHash(id compact.NodeID, h []byte, getSubtree GetSubtreeFunc) error {
```

Comment:
The tests for this function don't cover several of the branches. I suggest we look at that separately.

Reply:
Do you suggest a follow-up, or as a prereq for this PR?

Reply:
I found a simple way to add coverage here, will send a PR shortly.

Reply:
Followup is ok.

Reply:
Checked #2407 on this PR, tests pass. So submitting it optimistically.